src/main/java/com/zaxxer/hikari/pool/PoolBase.java

```diff
@@ -528,6 +528,9 @@ private void setLoginTimeout(final DataSource dataSource, final long connectionT
          catch (SQLException e) {
             LOGGER.warn("{} - Unable to set DataSource login timeout", poolName, e);
          }
+         catch (UnsupportedOperationException e) {
```

Comment:
As this is a pull request against *dev*, and *dev* is Java 7/8 only, please combine this catch with the above using a multi-catch statement.